tests/usermetadata/test_metadata.py

```diff
@@ -85,6 +87,66 @@ async def test_that_usermetadata_recipe_works_as_expected():
     assert get_metadata_res.metadata == {}
 
 
+@mark.asyncio
+async def test_usermetadata_recipe_shallow_merge():
+    init(
+        supertokens_config=SupertokensConfig('http://localhost:3567'),
+        app_info=InputAppInfo(
+            app_name='SuperTokens Demo',
+            api_domain='https://api.supertokens.io',
+            website_domain='supertokens.io'
+        ),
+        framework='fastapi',
+        recipe_list=[usermetadata.init()]
+    )
+    start_st()
+
+    version = await Querier.get_instance().get_api_version()
+    if get_max_version(version, '2.13.0') != version:
+        # If the version less than 2.13.0, user metadata doesn't exist. So skip the test
+        return
+
+    TEST_USER_ID = "userId"
+
+    TEST_METADATA_ORIGINAL: Dict[str, Any] = {
+        "updated": {
+            "subObjectNull": "this will become null",
+            "subObjectCleared": "this will be removed",
+            "subObjectUpdate": "this will become a number",
+        },
+        "cleared": "this should not be in the end result",
+    }
+
+    TEST_METADATA_UPDATE: Dict[str, Any] = {
+        "updated": {
+            "subObjectNull": None,
+            "subObjectUpdate": 123,
+            "subObjectNewProp": "this will appear",
+        },
+        "cleared": None,
+        "newRootProp": "this should appear in the end result"
+    }
+
+    TEST_METADATA_RESULT: Dict[str, Any] = {
+        "updated": {
+            "subObjectNull": None,
+            "subObjectUpdate": 123,
+            "subObjectNewProp": "this will appear",
+        },
+        "cleared": None,
```

Comment:
This should not be there in the result at all.

Reply:
@rishabhpoddar 
`==` works as expected. The problem was that `get_max_version` doesn't work the same for `2.13.0` and `2.13`. 
I've created a new function `is_version_lt` to avoid this confusion in the future.